rack-ui/pages/utility.py

```diff
@@ -0,0 +1,71 @@
+""" Content for the utility page """
+
+import time
+import platform
+from dash import html, dcc, callback, Input, Output, State
+import dash_bootstrap_components as dbc
+from .helper import *
+
+
+# dialog confirming triple store restart done
+restart_done_dialog = dbc.Spinner(dbc.Modal(
+    [
+        dbc.ModalBody("MESSAGE PLACEHOLDER", id="restart-done-dialog-body"),      # message
+        dbc.ModalFooter([
+            html.Button("Close", id="restart-done-button", n_clicks=0)            # close button
+        ]),
+    ],
+    id="restart-done-dialog",
+    is_open=False,
+    backdrop=False,
+))
+
+# page elements
+layout = html.Div([
+    html.H2('RACK Utilities'),
+    dcc.Markdown("_Utility functions for RACK_"),
+    html.Button("Restart triple store", id="restart-button", n_clicks=0),
+    restart_done_dialog
+])
+
+####### callbacks ######################################
+
+@dash.callback(
+    output=Output("restart-done-dialog-body", "children"),
+    inputs=Input("restart-button", "n_clicks"),                     # triggered by clicking restart button
+    background=True,                                                # background callback
+    running=[
+        (Output("restart-button", "disabled"), True, False),        # disable the button while running
+    ],
+    prevent_initial_call=True
+)
+def run_restart(n_clicks):
+    """
+    Restart the triple store
+    """
+    try:
+        if platform.system() == "Windows":
```

Comment:
It seems like we should probably only attempt to restart fuseki when RACK UI is running in our docker image. If the OS is Windows, then obviously we aren't running in our docker image, but a user running on MacOS or Linux running this command probably shouldn't be able to restart (or try to restart) fuseki this way. It likely won't work because they aren't using systemctl to run it (I'm not, for example) or because sudo will prompt for a password and will stall.

Is there some way we can distinguish if we're on Docker to avoid surprises here?